kpi/deployment_backends/mock_backend.py

```diff
@@ -118,6 +120,14 @@ class FakeRequest:
                 request=request,
             )
             if error:
+                error_dict = error.__dict__
+                if '_container' in error_dict and len(error_dict['_container']) > 0:
+                    xml_response = error_dict['_container'][0]
+                    message_text = DET.fromstring(xml_response).findtext(
+                        './/{http://openrosa.org/http/response}message'
+                    )
+                    if message_text == 'Instance ID is required':
+                        raise InstanceIdMissingError()
```

Comment:
Let's talk about this. It should be raised automatically without having this whole block

Reply:
In the test, we use `mock_submissions()`, which in turn calls `safe_create_instance()` for submission processing. If `safe_create_instance()` encounters an issue, it returns an `OpenRosaResponse` with an error code and message, rather than a specific exception class. This is why I added code to re-raise the specific exception based on the error content. However, I'm currently looking into it to improve this handling and make it more efficient